### PR TITLE
Fix bug: When IDs targetClass & model(not new) is same then unique validator passed

### PR DESCRIPTION
**Pull request by githubjeka**
Fix #10263

### CODE REVIEW

File: tests/framework/validators/UniqueValidatorTest.php [hunk 1]
```diff
@@ -136,4 +138,28 @@ public function testValidateCompositeKeys()
         $val->validateAttribute($m, 'id');
         $this->assertFalse($m->hasErrors('id'));
     }
+
+    public function testValidateTargetClass()
+    {
+        $val = new UniqueValidator([
+            'targetClass' => Category::className(),
+            'targetAttribute' => ['name'],
```
Comment (cebe) [inline tests/framework/validators/UniqueValidatorTest.php]: could you explain the semantics behind this validation? why would I validate uniqueness of item name against categories?

Comment (githubjeka) [inline tests/framework/validators/UniqueValidatorTest.php]: This is done so only for reproduction

Comment (githubjeka) [inline tests/framework/validators/UniqueValidatorTest.php]: I need two models with same attributes for unique

Comment (samdark) [inline tests/framework/validators/UniqueValidatorTest.php]: UUID?

Comment (samdark) [inline tests/framework/validators/UniqueValidatorTest.php]: Seems to be valid case.

**Comment by cebe** [comment]
needs a changelog

File: framework/CHANGELOG.md [hunk 2]
```diff
@@ -3,7 +3,7 @@ Yii Framework 2 Change Log
 
 2.0.7 under development
 -----------------------
-
+- Bug #10263: Unique validator passed for update-scenario when IDs targetClass and model is same (bupy7, githubjeka)
```
Comment (SilverFire) [inline framework/CHANGELOG.md]: We order our changelog by type (bug, enh, chg, new), then by issue ID.

**Comment by githubjeka** [comment]
@cebe  if need semantic I can change test. Because attribute can be different.

**Comment by SilverFire** [comment]
Merged with minor adjustments: https://github.com/yiisoft/yii2/commit/a1ff095f91f1da59305fd108f1bf0c063a554cb1, https://github.com/yiisoft/yii2/commit/6eb9830d645f9a00ac5568e695e209a761b226d2. Thank you!

**Pull request merged**